### PR TITLE
Fix Severity Rating Selection from Overview Bar Chart

### PR DESCRIPTION
**Pull request by chrtorres**
#  #

## 🗣 Description ##

- Fixed routing to Vulnerabilities table with selected filter applied from "Severity Levels" graph.
- Fixed onKeyDown not functional for 'Severity Levels', 'Most Common Ports', and 'Top 50 Vulnerable Domains' graphs.

## 💭 Motivation and context ##

Currently, when selecting a vulnerability rating such as Medium in the Severity Levels graph,  ‘All Vulnerabilities’ returns inconsistent data related to the severity level. With this fix, Vulnerabilities will be displayed with the selected view applied from the Severity Levels graph.

This resolves [CRASM-494](https://maestro.dhs.gov/jira/browse/CRASM-494)

## 📷 Screenshots ##

Screenshot 2024-08-23 at 9 51 27 AM

Screenshot 2024-08-23 at 9 53 01 AM

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [ ] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx [hunk 1]
```diff
@@ -197,27 +198,80 @@ export const Vulnerabilities: React.FC<{ groupBy?: string }> = ({
   const history = useHistory();
   const location = useLocation();
   const state = location.state as LocationState;
-  const [initialFilters, setInitialFilters] = useState<Filters<Vulnerability>>(
-    state?.title ? [{ id: 'title', value: state.title }] : []
+  // const paramsArray = [
```
Comment (ameliav) [inline frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx]: Unused code can be removed unless you write a comment in a line above it explaining why it should stay commented.

File: frontend/src/pages/Risk/TopVulnerableDomains.tsx [hunk 2]
```diff
@@ -70,6 +70,12 @@ const TopVulnerableDomains = (props: { data: Point[] }) => {
   //Custom Bar Layer to allow for top to bottom tab navigation
   const CustomBarLayer = ({ bars }: { bars: any[]; [key: string]: any }) => {
     const reversedBars = [...bars].reverse();
+    const filteredVulnTableLink =
+      '/inventory/vulnerabilities?sort=createdAt&desc=false';
```
Comment (ameliav) [inline frontend/src/pages/Risk/TopVulnerableDomains.tsx]: See https://github.com/cisagov/XFD/pull/555#discussion_r1729160116

**Review by ameliav — COMMENTED** [review]
See comments

File: frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx (hunk 1 above)
Comment (chrtorres) [reply frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx]: Good catch, this should have been removed. This has been resolved.

**Review by ameliav — APPROVED** [review]
LGTM

**Review by lwersiy — APPROVED** [review]
LGTM

**Review by jayjaybunce — APPROVED** [review]
Looks good, functionality feels great.

**Review by hawkishpolicy — APPROVED** [review]
LGTM.

**Review by schmelz21 — APPROVED** [review]
Merging

**Pull request merged**